### PR TITLE
chore: unit tests for credit service

### PR DESCRIPTION
**Pull request by anujk14**

### CODE REVIEW

File: billing/credit/service.go [hunk 1]
```diff
@@ -32,15 +32,21 @@ func (s Service) Add(ctx context.Context, cred Credit) error {
 		return fmt.Errorf("credit amount is negative")
 	}
 	// check if already credited
-	if t, err := s.transactionRepository.GetByID(ctx, cred.ID); err == nil && t.ID != "" {
+	t, err := s.transactionRepository.GetByID(ctx, cred.ID)
+
+	if err != nil {
```
Comment (kushsharma) [inline billing/credit/service.go:37]: this will fail to add transaction if it returns not found from repository for this id. We want to do opposite of it.

Comment (anujk14) [reply billing/credit/service.go]: Yup, missed that. Added a check for it.

File: billing/credit/service.go [hunk 2]
```diff
@@ -32,15 +32,21 @@ func (s Service) Add(ctx context.Context, cred Credit) error {
 		return fmt.Errorf("credit amount is negative")
 	}
 	// check if already credited
-	if t, err := s.transactionRepository.GetByID(ctx, cred.ID); err == nil && t.ID != "" {
+	t, err := s.transactionRepository.GetByID(ctx, cred.ID)
+
+	if err != nil && err != ErrNotFound {
```
Comment (kushsharma) [inline billing/credit/service.go:37]: better we use `errors.Is` instead of != check to make sure it unwraps correctly?

Comment (anujk14) [reply billing/credit/service.go]: Done

**Review by kushsharma — APPROVED** [review]

**Pull request merged**